parcels/particlefile.py

```diff
@@ -125,25 +125,25 @@ def write(self, pset, time, sync=True, deleted_only=False):
            (self.write_ondelete is False or deleted_only is True):
             if pset.size > 0:
 
-                first_write = [p for p in pset if p.fileid < 0]
+                first_write = [p for p in pset if p.fileid < 0 or len(self.idx) == 0]  # len(self.idx)==0 in case pset is written to new ParticleFile
                 for p in first_write:
                     p.fileid = self.lasttraj
                     self.lasttraj += 1
 
-                inds = [p.fileid for p in pset]
-
-                self.id[inds, self.idx] = [p.id for p in pset]
-                self.time[inds, self.idx] = time
-                self.lat[inds, self.idx] = np.array([p.lat for p in pset])
-                self.lon[inds, self.idx] = np.array([p.lon for p in pset])
-                self.z[inds, self.idx] = np.array([p.depth for p in pset])
-                for var in self.user_vars:
-                    getattr(self, var)[inds, self.idx] = np.array([getattr(p, var) for p in pset])
-                for var in self.user_vars_once:
-                    if np.any(first_write):
-                        vals = [getattr(p, var) for p in first_write]
-                        newinds = [p.fileid for p in first_write]
-                        getattr(self, var)[newinds] = np.array(vals)
+                self.idx = np.append(self.idx, np.zeros(len(first_write)))
+
```

Comment:
Nice development! It's great to indeed have the nc files written by Parcels as small and simple as possible.

But it seems that we don't document anywhere what is the format of those files. Shouldn't we do it?
For example, I received an email from @Jacketless yesterday that had problems to read an old file generated by Parcels in the new code. It may be related to the commit with arrays and indexed pfiles.
Should we explain the file format somewhere?